lib/typeprof/core/env/module_entity.rb

```diff
@@ -350,6 +354,7 @@ def on_ancestors_updated(genv, base_mod)
         end
       end
       @ivar_reads.each {|ivar_read| genv.add_run(ivar_read) }
+      @cvar_reads.each {|cvar_read| genv.add_run(cvar_read) }
```

Comment:
I'm just adding it without really understanding what it means, but is it ok?

Reply:
I think this change is good.

`on_ancestors_updated` is a hook that is called when the superclass of a class changes.
Consider the following example, and you remove the definition of `Foo::C`.

```ruby
class C
  @@foo = 1
end

module Foo
  # if this definition of `C` is removed,
  class C
    @@foo = "str"
  end

  class D < C # the meaning of `C` changes from `::D::C` to `::C`
    p @@foo # and this cvar also changes from `::D::C`'s `@@foo` to `::C`'s `@@foo`
  end
end
```

The superclass `C` in `class D < C` will change its meaning from `::D::C` to `::C`. In this case, we need to re-evaluate `@@foo`.

The code that you added enqueues all cvar reads in the class into the revaluation queue.